src/main/java/com/salesforce/cdp/queryservice/interceptors/MetadataCacheInterceptor.java

```diff
@@ -16,35 +16,74 @@
 
 package com.salesforce.cdp.queryservice.interceptors;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.salesforce.cdp.queryservice.core.QueryServiceConnection;
+import com.salesforce.cdp.queryservice.model.MetadataCacheKey;
 import com.salesforce.cdp.queryservice.util.Constants;
-import com.salesforce.cdp.queryservice.util.MetadataCacheUtil;
 import lombok.extern.slf4j.Slf4j;
-import okhttp3.*;
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
 import org.apache.http.HttpStatus;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.util.concurrent.TimeUnit;
 
 @Slf4j
 public class MetadataCacheInterceptor implements Interceptor {
+    private final QueryServiceConnection connection;
+    private Cache<MetadataCacheKey, String> metaDataCache;
+
+    public MetadataCacheInterceptor(QueryServiceConnection connection) {
+        this.connection = connection;
```

Comment:
We dont need connection in this interceptor

Reply:
So can I initialize/pass in to interceptor with the cacheKey which is combination of tenantUrl + dataspace?

Reply:
Removed the connection dependency in metadata cache interceptor.